data/tutorials/guides/1wf_06_database.md

```diff
@@ -0,0 +1,328 @@
+
+There are different types of queries. A `get_many` query (like the example) returns a list of
+rows. `get_one` just returns the unique row, `get_opt` returns an option value (`None` or `Some`
+depending of the existance of a queried row), `execute` returns nothing `()`.
+
+By default rows are presented as tuples.
+
+### ppx_rapper options
+
+Depending of the needs, a `ppx_rapper` query can be written as
+
+```ocaml
+  [%rapper get_many {sql|...|sql} options]
+```
+
+Where supported options are:
+
+- `record_in` : instead of calling the query with one argument per parameter, a single
+argument is provided. It must be a record whose the fields correspond to the parameters.
+- `record_out` : instead of outputing row(s) as tuple(s), they are presented as record(s).
+- `function_out` : the query is called with a function as its first argument. This function
+is called for each row with every values (column names) as a labelled arguments. The return value of the query
+is composed of the returned values of this function. This works like [List/Option.map].
+
+Options are separated by whitespaces.
+
+## Custom error handling
+
+`(let*)` from `Lwt_result.Syntax` is handy if we want a default error
+handling (stopping the chain of queries, returning an error type).
+
+If we want our own error handling, we can use the `lwt_ppx` preprocessor and
+use the `let%lwt` or `match%lwt` tokens.
+
+Typically:
+
+```ocaml
+     match%lwt query_a () cnx with
+     | Result.Ok _value -> do_something;
+                  Lwt.return (Result.Ok ())
+     | Result.Error err -> do_something;
+                  Lwt.return (Result.Error err)
+```
+
+Note: we can also change the `Lwt.return` values... An ok query could be
+wrong if the returned values do not respect some assertions. An error query could also be mitigated.
+
+If we want to embed the custom error handled query in the `(let*)` chain,
+simply type:
+
+```ocaml
+ let* () = match%lwt ... in
+```
+
+## Itering a list of queries, transactions
+
+Using `List.map` can easily create queries... but they won't be chained from
+a `lwt` point of vue. The `Lwt_list.iter_s` and `Lwt_list.map_s` can iter
+them... but won't stop at the first error.
+
+If we wan't to iter all the queries even in the presence of error, we can
+simply type
+
+```ocaml
+let%lwt () = Lwt_list.iter_s  item_to_query_function list in
+Lwt.return (Result.OK ())
+```
+
+Or
+
+```ocaml
+let%lwt result_list = Lwt_list.map_s item_to_query_function list in
+analyse_the_result_list;
+Lwt.return (Result.OK ())
+```
+
+But it would be better to define a function:
+
+```ocaml
+let iter_queries queries cnx =
+   List.fold_left
+     (fun a f -> Lwt_result.bind a (fun () -> f cnx))
+     (Lwt.return (Ok ()))
+     queries
+```
+
+This function iter a list of functions with a connexion as an argument and execute each query.
+It can be used in the following context (where `insert_a_record` is defined in the example
+section):
```

Comment:
```suggestion
This function iters a list of functions with a connection as an argument and executes each query.
It can be used in the following context (where `insert_a_record` is defined in the example
section):
```